docs/nlu/training-data-format.rst

```diff
@@ -32,7 +35,7 @@ e.g. ``[entity](entity name)``.
 
     ## intent:greet
     - hey
-    - hello
+    - [greet[hello]] <!-- keyword for intent greet>
```

Comment:
I'm not quite sure i understand why we need this keyword... It seems to me like you should just be able to add it as normal in a list?

Reply:
I chose this because it allows users to specify keywords in existing examples while also adding them individually (just add example with only the keyword). The specific format `[intent[keyword]]` is arbitrary, this way its quite simple while also being unlikely to be something that anyone already has in their examples.
What do you mean by adding it as normal in a list?

Reply:
sorry i just meant that if we allow people to specify keywords like this:
```
 ## intent:greet
    - hey
    - hello
```
There's no need for the `[intent[keyword]]` format right?

Reply:
Yes, but then every word in the examples would be a keyword? Or every example that contains only one word?

Reply:
I assumed you were already allowing people to specify it like that in the list? I guess i'm confused about the case you state where you don't label the intent :D

Reply:
Ah sorry, I'll try to make the docs clearer
No that is not the case, only words are that in double brackets are keywords. The difference between `[[keyword]]` and `[intent[keyword]]` is that the former is assumed to be a keyword for the intent under which it is found, while the in the latter case the keyword is for the specified intent no matter under which intent the example is given.
The second style was inspired by a forum post asking for a way to filter out cussing. I think it could be useful when using multiple intents.

Generally the way I picture this classifier being useful is A) for people to get going when they have few training data and B) to avoid missing crucial words (e.g. "stolen"). Based on that the special marking for keywords is there to allow people to use this in combination with a machine-learning based classifier.

Reply:
sorry for dropping this so long 🙈 i don't think we can have two intent classifiers anyways, so in my opinion we should just use the standard list approach for markdown. But I'd also like to get someone else's input on this -- @Ghostvv do you have any thoughts on the format of providing the data to the intent classifier?

Reply:
hm... the whole thing is rather tricky especially if keyword doesn't correspond to current intent. Also how do you deal with multi classifier prediction?

Reply:
to me original issue description more correspond to keyword entity extractor

Reply:
what? no this is about a keyword intent classifier, not entities. Also what do you mean multi classifier prediction? We don't do that right now. 

And what do you mean with this "the whole thing is rather tricky especially if keyword doesn't correspond to current intent"

This is just supposed to be a super simple standalone keyword classifier

Reply:
if it is standalone, why do we need special train data format?

Reply:
You can just add two classifiers to the pipeline. Usually this does nothing as they are called sequentially and just set a message property, so the second always overrides the first. However with the keyword intent classifier it only sets the message intent if it finds a keyword. That way it can be used next to any other classifier.

It is also good to be able to use it together with others. 1) It is useless for anything nontrivial 2) if somebody actually starts building a both with this thing and then want to transition to ML classifiers this would be much easier if they can use both 3) having 100% classification might be something people want even for proper bots (e.g. they just want to feel sure not to miss some super important word ever)

If users provide NLU data in the "normal" format, instead of literally providing lists of keywords where the examples usually go, we will need some way of indicating what the keywords are.

Reply:
can we have smith like `keywords:` under `## intent:`? using brackets seem very complicated to me especially if keyword intent is different, then this datapoint is useless because keyword classifier will override it

Reply:
We could move it to the domain file yes, but can policy access domain while training or predicting?

Why would it override it in training though? Simply because the name of an intent is in the example? In that case we should definitely remove that feature and only have keywords as examples for the intent to which they belong.

Reply:
So we did solve this finally :sweat_smile: